lib/pharos/host/ubuntu/scripts/upgrade-kubeadm.sh

```diff
@@ -2,12 +2,15 @@
 
 set -ex
 
-if [ $(kubeadm version -o short) = "v${VERSION}" ]; then
+if [ -x /usr/local/bin/pharos-kubeadm-${VERSION} ]; then
     exit
 fi
 
-cd /tmp
-export DEBIAN_FRONTEND=noninteractive
-apt-get download kubeadm=${VERSION}-00
-dpkg -i --ignore-depends=kubelet kubeadm_${VERSION}*.deb
-rm -f kubeadm_${VERSION}*.deb
\ No newline at end of file
+BIN_URL="https://dl.bintray.com/kontena/pharos-bin/kube/${VERSION}/kubeadm-${ARCH}.gz"
+
+curl -fsSL $BIN_URL -o /tmp/kubeadm.gz
```

Comment:
We need to curl binary because kubeadm decided to depend on `cri-tools`?

Reply:
No, it's because upgrading the `kubeadm` package will upgrade the `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` to reference the `/var/lib/kubelet/config.yaml` and place the kubelet into a crash-and-restart loop.

I don't seem to have copied the exact error, but IIRC this somehow prevented the `kubeadm upgrade host config` from working, deadlocking the upgrade.